db/model/Gdoc/htmlToEnriched.ts

```diff
@@ -86,82 +95,966 @@ export function htmlToSimpleTextBlock(html: string): EnrichedBlockSimpleText {
+                    }
+                }
+            )
+            .with(
+                { tagName: P.union("svg", "table", "video") },
+                (): BlockParseResult<EnrichedBlockHtml> => {
+                    return {
+                        errors: [],
+                        content: [
+                            {
+                                type: "html",
+                                value: $.html(element) ?? "",
+                                parseErrors: [],
+                            },
+                        ],
+                    }
+                }
+            )
+            .otherwise(() => ({
+                errors: [
+                    {
+                        name: "unhandled html tag found",
+                        details: `Encountered the unhandled tag ${element.tagName}`,
+                    },
+                ],
+                content: [],
+            }))
+        return result
+    } else
+        return {
+            errors: [
+                {
+                    name: "unkown element tag",
+                    details: `type was ${element.type}`,
+                },
+            ],
+            content: [],
+        }
+}
+
+export function cheerioElementsToArchieML(
+    elements: CheerioElement[],
+    $: CheerioStatic
+): BlockParseResult<ArchieBlockOrWpComponent> {
+    if (!elements)
+        return {
+            errors: [],
+            content: [],
+        }
+    let remainingElements: CheerioElement[] = elements
+    const parsedContent: BlockParseResult<ArchieBlockOrWpComponent>[] = []
+    while (remainingElements.length > 0) {
+        const element = remainingElements[0]
+        if (isWpComponentStart(element)) {
+            const parseResult = parseWpComponent(remainingElements, $)
+            parsedContent.push(parseResult.result)
+            remainingElements = parseResult.remainingElements
+        } else if (element.type === "comment") {
+            remainingElements = remainingElements.slice(1)
+        } else {
+            const parsed = cheerioToArchieML(element, $)
+            const cleaned = withoutEmptyOrWhitespaceOnlyTextBlocks(parsed)
+            if (cleaned.content.length > 0 || cleaned.errors.length > 0)
+                parsedContent.push(parsed)
+            remainingElements = remainingElements.slice(1)
+        }
+    }
+    return joinBlockParseResults(parsedContent)
+}
+
+export function withoutEmptyOrWhitespaceOnlyTextBlocks(
+    result: BlockParseResult<ArchieBlockOrWpComponent>
+): BlockParseResult<ArchieBlockOrWpComponent> {
+    return {
+        ...result,
+        content: result.content.filter(
+            (element) =>
+                !("type" in element) ||
+                ("type" in element && element.type !== "text") ||
+                (element.value.length > 0 &&
+                    element.value.some(
+                        (span) =>
+                            span.spanType !== "span-simple-text" ||
```

Comment:
This function took me a second to wrap my head around but does this mean we're removing any text element if it has a child that isn't a `span-simple-text`?

Reply:
simplified this code a little